Player-OnQuit-BugFixing

uzaiiirahmed opened the pull request:
Following are the updates in this fork :-
Enabled 4 playes Support for 2d-platform-scene.
Count for current players in the lobby.
feature to send string using RPC on all connected player.
feature to send custom data types using RPC

uzaiiirahmed committed:
feat : Enable 4 players and update build settings

Increased max players per room to 4 and updated score text assignment logic in GameManager2d.cs. Changed mockMode and browser settings in platformer and lobby scenes. Updated build settings to enable 2d-platformer and TestScene, and switched WebGL template to PWA. Added .vsconfig for Visual Studio workload and made minor .vscode/settings.json adjustments.

uzaiiirahmed committed:
feat : Add player count UI to 2D platformer example

Introduced a PlayerCountText UI element to display the current number of players in the 2D platformer example. Updated GameManager2d to manage and refresh the player count display when players join or leave.

uzaiiirahmed committed:
feat : Hello World RPC display test to 2d platformer

Introduced a new TextMeshProUGUI field and UI element for displaying 'Hello, World!' in the 2d platformer example. Added RPC registration and logic to show the message on all players' screens when triggered, with automatic clearing after a delay.

uzaiiirahmed committed:
feat: RPC for custom data transfer in 2D platformer

Introduced new RPC methods to send and receive custom data types as JSON strings between players. Added IdkCoolClass for demonstration, updated GameManager2d to handle custom data RPCs, and included input handling for triggering these actions. Minor formatting and whitespace adjustments in PlayerController2d.

uzaiiirahmed committed:
refactor: Improve RPC data handling

Enhanced RPC.cs to serialize custom serializable types using JsonUtility. Refactored GameManager2d.cs for better RPC registration, error handling, and data transmission. Updated .vscode and project settings for new solution and WebGL template. Disabled browser in 2d-platformer example scene.

HamadUllah16 committed:
feat: Enhance RPC data handling with additional logging and new data structure

HamadUllah16 committed:
refactor: Simplify data parsing in RPC by using null-conditional operator

uzaiiirahmed committed:
Merge pull request #1

Player-OnQuit-BugFixing

uzaiiirahmed committed:
chore : re-added chrome drivers

momintlh commented on .vscode/settings.json:
Let's not push vscode settings, we should just remove .vscode folder and add it to `gitignore` but for now just undo this change.

momintlh commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -77,32 +84,40 @@ void Start()
+        // Register Rpc for hello world display
+        _playroomKit.RpcRegister("DisplayHelloWorldRPC", (data, caller) => DisplayHelloWorldRPC(), "Displays Hello World on all players' screens");
+
+        // Register Rpc for sending an d recieving custom data types 
```
cleanup some of these comments

uzaiiirahmed replied:
Done

momintlh commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -77,32 +84,40 @@ void Start()
     /// <summary>
     /// Update the Score UI of the player and sync.
     /// </summary>
     void HandleScoreUpdate(string data, string caller)
```
Why was this updated?

momintlh commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -187,11 +248,138 @@ private static void RemovePlayer(string playerID)
     }
+
+    /// <summary>
+    /// Methode the player count text UI.
```
spelling mistake here

momintlh commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -187,11 +248,138 @@ private static void RemovePlayer(string playerID)
+
+
+    // RPC method to display "Hello World" on all players' screens
+    private void DisplayHelloWorldRPC()
```
Signature should be `(string senderId, string data)`

momintlh commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -187,11 +248,138 @@ private static void RemovePlayer(string playerID)
+
+    private void SendCoolData()
+    {
+     IdkCoolClass coolData = new IdkCoolClass
```
Something like ` class PlayerData`

momintlh commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -187,11 +248,138 @@ private static void RemovePlayer(string playerID)
+            Debug.Log("RPC call to send cool data sent successfully.");
+        });
+    
+          _playroomKit.RpcCall("ReceiveCoolData", "test message", PlayroomKit.RpcMode.ALL, () =>
```
Remove this one, goal here is to only test if a class object is being serialized and received by others

uzaiiirahmed committed:
feat : Player handling dynamically.

Adding and removing of players is handled dynamically now also scores are handled and the color are fetched from

uzaiiirahmed committed:
refactor :RemovePlayer to non-static

Changed RemovePlayer from static to instance method and removed redundant instance lookup. Streamlined player removal logic and direct access to class members for improved clarity and maintainability.

uzaiiirahmed commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -187,11 +248,138 @@ private static void RemovePlayer(string playerID)
+        }
+    }
+
+    private System.Collections.IEnumerator ClearHelloWorldAfterDelay(float delay)
```
so it doesnt stays on screen during the whole process of test

uzaiiirahmed committed:
feat : hello world now actually transfer from P2P

Updated the DisplayHelloWorldRPC method to accept data and caller parameters, allowing custom messages to be displayed. The RPC call now sends 'Hello, World!' as data, and the text display logic uses the provided data instead of a hardcoded string.

uzaiiirahmed committed:
feat : TestDictionaryRPC for sending and receiving structured data

Introduces TestDictionaryData and NestedData classes for structured data transfer via RPC. Registers a new 'TestDictionaryRPC' method, sends test data on 'P' key press, and adds a handler to deserialize and log the received data.

uzaiiirahmed committed:
chore : updated Ui and re added hello world on H key

uzaiiirahmed commented on Assets/PlayroomKit/modules/RPC/RPC.cs:
```diff
@@ -155,9 +168,18 @@ private static string ConvertToJson(object data)
                 else
                 {
-                    return ConvertComplexToJson(data);
+                    var result = ConvertComplexToJson(data);
```
done

uzaiiirahmed committed:
chore : removed unnecessary $ symb.

momintlh commented on Assets/PlayroomKit/Examples/2d-platformer/scripts/GameManager2d.cs:
```diff
@@ -6,57 +6,51 @@
-    private string newData;
-
+    [System.Serializable] public class ScoreData { public int value; } 
+    [System.Serializable]public class TestDictionaryData{
```
Change name to TestData and keep it. but also test sending dictionary on RPC separately.

uzaiiirahmed committed:
feat :Improved RPC data handling and add dictionary RPC support

Refactored GameManager2d to distinguish between TestData and dictionary-based RPC calls, adding new input handling for sending both types. Updated RPC.cs to better serialize/deserialize complex and dictionary data, improving error handling and JSON conversion. Updated .gitignore to exclude .vsconfig and .vscode/settings.json.
